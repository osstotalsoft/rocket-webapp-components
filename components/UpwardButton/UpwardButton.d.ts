export interface UpwardButtonProps {
    /**
     * The text displayed when user hover over the button
     */
    title?: string;
    /**
     * Callback fired when a "click" event is detected
     * @param {object} event The event source of the callback.
     */
    onClick: (event: React.SyntheticEvent) => void
    /**
     * If set to true, the button will be disabled
     */
    disabled?: boolean
    /**
    * @default "theme" 
    * The color of the button
     */
    color?: "primary" |
            "info" |
            "theme" |
            "themeNoBackground" |
            "themeWithBackground" |
            "success" |
            "warning" |
            "danger" |
            "rose" |
            "white" |
            "simple" |
            "defaultNoBackground" |
            "primaryNoBackground" |
            "infoNoBackground" |
            "successNoBackground" |
            "warningNoBackground" |
            "dangerNoBackground" |
            "roseNoBackground"
    /**
     * The size of the button
     * @default "medium"
     */
    size?: "small" |
             "medium"
    /**
    * @default "small" 
    * The fontSize of button's content.
    */
    fontSize?: "inherit" |
                "small" |
                "medium" |
                "large"
    /**
    * Override or extend the styles applied to the component
    */
    customClass?: string
    /**
    * @default "false"
    * If true, the keyboard focus ripple will be disabled. 
    */
    disableFocusRipple?: boolean
    /**
     * If true, the ripple effect will be disabled.
     */
    disableRipple?: boolean
    /**
     * @default "false"
     * If given, uses a negative margin to counteract the padding on one side 
     * (this is often helpful for aligning the left or right side of the icon 
     * with content above or below, without ruining the border size and shape).
     */
    edge?: "start" | "end" | "false"

    /**
    * Demos:
    * 
    * - https://bit.dev/totalsoft_oss/react-mui/upward-button
    * 
    *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/UpwardButton/UpwardButton.js
    * 
    * Bit component used:
    * 
    * - IconButton: https://material-ui.com/api/icon-button/
    *
    */

}

export default function UpwardButton(props: UpwardButtonProps): JSX.Element;