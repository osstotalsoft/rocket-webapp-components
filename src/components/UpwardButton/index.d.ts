export { default } from './UpwardButton';
export * from './UpwardButton';