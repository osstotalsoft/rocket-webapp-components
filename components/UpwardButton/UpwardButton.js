import React from "react";
import PropTypes from "prop-types";
import { ArrowUpward } from '@material-ui/icons';
import CustomIconButton from '../IconButton';

function UpwardButton({ title, onClick, disabled, fontSize, ...rest }) {
    return (
        <CustomIconButton aria-label="Upward" disabled={disabled} onClick={onClick} tooltip={title} {...rest}>
            <ArrowUpward fontSize={fontSize} />
        </CustomIconButton>
    )
}

UpwardButton.defaultProps = {
    color: "theme",
    fontSize: "small",
    size: "medium"
};

UpwardButton.propTypes = {
    title: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground"
    ]),
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    fontSize: PropTypes.oneOf([
        "inherit",
        "small",
        "medium",
        "large"
    ])
}

export default UpwardButton;