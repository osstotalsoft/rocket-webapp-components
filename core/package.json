{
  "engines": {
    "node": ">=10"
  },
  "scripts": {
    "prebuild": "rimraf build",
    "build": "npm run build:cjs &&  npm run build:copy-files",
    "build:cjs": "babel src -d build --copy-files",
    "build:copy-files": "node ./scripts/copy-files.js",
    "bit-build": "bit -v && bit import && bit build"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.0",
    "@babel/core": "^7.7.7",
    "@babel/node": "^7.7.7",
    "@babel/plugin-proposal-decorators": "^7.8.3",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-react": "^7.8.0",
    "babel-eslint": "^10.0.3",
    "cross-env": "^7.0.2",
    "eslint": "^6.8.0",
    "eslint-config-prettier": "^6.7.0",
    "eslint-config-standard": "^14.1.0",
    "eslint-config-standard-react": "^9.2.0",
    "eslint-plugin-import": "^2.18.2",
    "eslint-plugin-node": "^11.0.0",
    "eslint-plugin-prettier": "^3.1.1",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-react": "^7.17.0",
    "eslint-plugin-standard": "^4.0.1",
    "gh-pages": "^2.2.0",
    "npm-run-all": "^4.1.5",
    "prettier": "^1.19.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "rimraf": "^3.0.0",
    "url-search-params-polyfill": "^7.0.1"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-flags": "^0.1.18",
    "react-router-dom": "^5.1.2"
  },
  "sideEffects": false,
  "dependencies": {},
  "bit": {
    "env": {
      "compiler": "totalsoft_oss.react-mui/compilers/react@1.0.32"
    },
    "componentsDefaultDirectory": "components/{name}",
    "packageManager": "npm"
  }
}
