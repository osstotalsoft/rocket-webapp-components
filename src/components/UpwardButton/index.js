import UpwardButton from "./UpwardButton";
export default UpwardButton;