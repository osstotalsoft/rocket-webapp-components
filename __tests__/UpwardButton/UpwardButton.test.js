import React from "react";
import UpwardButton from "../../components/UpwardButton/UpwardButton";
import { mount } from 'enzyme';
import { IconButton } from "@material-ui/core";

describe("UpwardButton", () => {
  it("onClick should be called", () => {
    const onClickMock = jest.fn();
    const wrapper = mount(<UpwardButton
      onClick={onClickMock}
      title="Upward" />);

    wrapper.find(IconButton).simulate('click')
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('the button should be disabled', () => {
    const onClickMock = jest.fn();

    const wrapper = mount(<UpwardButton
      onClick={onClickMock}
      disabled={true}
      title="Upward" />);

    expect(wrapper.find(IconButton).props().disabled).toBe(true);
  });

  it('the button should be enabled', () => {
    const onClickMock = jest.fn();

    const wrapper = mount(<UpwardButton
      onClick={onClickMock}
      disabled={false}
      title="Upward" />);

    expect(wrapper.find(IconButton).props().disabled).toBe(false);
  });

  it('the color, fontSize and size props should be equal to the default values if they are not provided', () => {
    const onClickMock = jest.fn();

    const wrapper = mount(<UpwardButton
      onClick={onClickMock}
      disabled={false}
      title="Upward" />);

    expect(wrapper.props().color).toBe('theme');
    expect(wrapper.props().fontSize).toBe('small');
    expect(wrapper.props().size).toBe('medium');
  });
});
