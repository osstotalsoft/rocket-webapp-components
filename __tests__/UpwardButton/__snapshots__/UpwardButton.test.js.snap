// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpwardButton Snapshot 1`] = `
<span
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Upward"
>
  <button
    aria-label="Upward"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</span>
`;
